### PR TITLE
[ROCm] NonMaxSuppression

### PR DESCRIPTION
**Pull request by ekuznetsov139**
This PR fixes and enables the NonMaxSuppression kernel and corresponding test for ROCm.

### CODE REVIEW

**Comment by gbaned** [comment]
@chsigg Can you please take a look on this PR ? Thanks!

File: tensorflow/core/kernels/non_max_suppression_op.cu.cc [hunk 1]
```diff
@@ -28,26 +28,11 @@ limitations under the License.
 #include "tensorflow/core/util/gpu_launch_config.h"
 #include "tensorflow/stream_executor/stream_executor.h"
 
-#define TF_RETURN_IF_CUDA_ERROR(result)                   \
```
Comment (cheshire) [inline tensorflow/core/kernels/non_max_suppression_op.cu.cc:31]: Why are these macros removed in this PR?

Comment (ekuznetsov139) [reply tensorflow/core/kernels/non_max_suppression_op.cu.cc:31]: They are now in core/util/gpu_device_functions.h (as of PR #36641). The ones in non_max_suppression_op.cu.cc are duplicates.

Comment (cheshire) [reply tensorflow/core/kernels/non_max_suppression_op.cu.cc:31]: Then can we remove them in a separate PR?

Comment (ekuznetsov139) [reply tensorflow/core/kernels/non_max_suppression_op.cu.cc:31]: I suppose

**Comment by gbaned** [comment]
@ekuznetsov139  Can you please resolve conflicts? Thanks!

**Comment by chsigg** [comment]
This PR has conflicts and doesn't build. Could you please address the issues? Thanks.

**Comment by gbaned** [comment]
@chsigg Can you please take a look on this PR ? Thanks!

**Review by chsigg — APPROVED** [review]

**Pull request merged**